src/cpp/src/sequence_group.hpp

```diff
@@ -425,59 +425,46 @@ class SequenceGroup {
         return m_generation_stream->get_status() == GenerationStatus::DROPPED_BY_HANDLE;
     }
 
-    void notify_handle() {
+    void push_outputs() {
+        GenerationOutputs outputs;
+        for (auto& sequence: m_sequences) {
+            GenerationOutput output;
+            output.generated_token_ids = sequence->get_generated_ids();
+            output.score = sequence->get_beam_search_score(m_sampling_params);
```

Comment:
beam search score can be used only for beam search, while this method `push_outputs` is used for greedy and multinomial as well. For non-beam search we need to use cumulative logprob.

Reply:
You're right, I'm already fixing that in my next PR